protocol-definitions/custom/Custom.yaml

```diff
@@ -1151,3 +1151,288 @@ customTypes:
+        nullable: false
+        since: 2.7
+      - name: kubernetesConfig
+        type: KubernetesConfig
+        nullable: false
+        since: 2.7
+      - name: eurekaConfig
+        type: EurekaConfig
+        nullable: false
+        since: 2.7
+      - name: discoveryConfig
+        type: DiscoveryConfig
+        nullable: false
+        since: 2.7
+      - name: syncConfig
+        type: WanSyncConfig
+        nullable: false
+        since: 2.7
+      - name: endpoint
+        type: String
+        nullable: true
+        since: 2.7
+  - name: AwsConfig
+    since: 2.7
+    params:
+      - name: tag
+        type: String
+        nullable: false
+        since: 2.7
+      - name: enabled
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: usePublicIp
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: properties
+        type: Map_String_String
+        nullable: false
+        since: 2.7
+  - name: GcpConfig
+    since: 2.7
+    params:
+      - name: tag
+        type: String
+        nullable: false
+        since: 2.7
+      - name: enabled
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: usePublicIp
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: properties
+        type: Map_String_String
+        nullable: false
+        since: 2.7
+  - name: AzureConfig
+    since: 2.7
+    params:
+      - name: tag
+        type: String
+        nullable: false
+        since: 2.7
+      - name: enabled
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: usePublicIp
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: properties
+        type: Map_String_String
+        nullable: false
+        since: 2.7
+  - name: KubernetesConfig
+    since: 2.7
+    params:
+      - name: tag
+        type: String
+        nullable: false
+        since: 2.7
+      - name: enabled
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: usePublicIp
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: properties
+        type: Map_String_String
+        nullable: false
+        since: 2.7
+  - name: EurekaConfig
+    since: 2.7
+    params:
+      - name: tag
+        type: String
+        nullable: false
+        since: 2.7
+      - name: enabled
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: usePublicIp
+        type: boolean
+        nullable: false
+        since: 2.7
+      - name: properties
+        type: Map_String_String
+        nullable: false
+        since: 2.7
+  - name: DiscoveryStrategyConfig
+    since: 2.7
+    params:
+      - name: className
+        type: String
+        nullable: false
+        since: 2.7
+      - name: properties
+        type: Map_String_Data
+        nullable: false
+        since: 2.7
+  - name: DiscoveryConfig
```

Comment:
I think we should name the following like XXConfigHolder instead of XXXConfig: 

"DiscoveryConfig",
    "DiscoveryStrategyConfig",
    "WanSyncConfig",

Reply:
Comment captured in later comment.